pymsis/utils.py

```diff
@@ -92,19 +150,34 @@ def _load_f107_ap_data() -> dict[str, npt.NDArray]:
     # so we can't just go back in line lengths)
     with _F107_AP_PATH.open() as fin:
         with BytesIO() as fout:
+            last_obs_date: np.datetime64 | None = None
             for line in fin:
-                if "PRM" in line:
-                    # We don't want the monthly predicted values
-                    continue
-                if ",,,,,,,," in line:
-                    # We don't want lines with missing values
+                if "PRM" in line or ",,,,,,,," in line:
+                    # We don't want the monthly predicted values or missing values
                     continue
+                if ",OBS," in line:
+                    # Capture last observed date
+                    last_obs_date = np.datetime64(line.split(",")[0])
```

Comment:
Can we get this from the `_DATA` numpy arrays? Maybe `warn_data` contains this information already and we could do something like (have not tried this) `_DATA["dates"][~_DATA["warn_data"]][-1]`.